hydra-protocol/Hydra/Protocol/Figures/fig_offchain_prot.tex

```diff
@@ -84,8 +84,16 @@
 						\On{$(\hpRS,v,s,\mT_{\mathsf{req}}, \tx_{\omega})$ from $\party_j$}{
 							\Req{$v = \hatv ~ \land ~ s = \hats + 1 ~ \land ~ \hpLdr(s) = j$} \;
 							\Wait{$\hats = \bar{\mc S}.s$}{
-								\Req{$\bar{\mc S}.U \applytx \tx_{\omega} \not= \bot$} \;
-								$U_{\mathsf{active}} \gets \bar{\mc S}.U \applytx \tx_{\omega} \setminus \mathsf{outputs(\tx_{\omega})}$ \;
+                                \If{$v = \bar{\mc S}.v ~ \land ~ \bar{\mc S}.\tx_{\omega} \neq \bot$ }{
```

Comment:
Must: port this correctly!

This was the latest version of this section on the branch we worked together:

```
							\Req{$s = \hats + 1 ~ \land ~ \hpLdr(s) = j$} \;
							\Wait{$ v = \hatv ~ \land ~ \hats = \bar{\mc S}.s$}{
							    % TODO: waiting for version observed is way longer than for no snapshot in flight!
								\If{$v = \bar{\mc S}.v ~ \land ~ \bar{\mc S}.\tx_{\omega} \neq \bot$ }{
									\Req{$\bar{\mc S}.\tx_{\omega} = \tx_{\omega}$} \;
									$U_{\mathsf{active}} \gets \bar{\mc S}.U$ \;
									$U_{\omega} \gets \bar{\mc S}.U_{\omega}$
								}
								\Else{
									\Req{$\bar{\mc S}.U \applytx \tx_{\omega} \not= \bot$} \;
									$U_{\mathsf{active}} \gets \bar{\mc S}.U \applytx \tx_{\omega} \setminus \mathsf{outputs(\tx_{\omega})}$ \;
									$U_{\omega} \gets \mathsf{outputs}(\tx_{\omega})$
								
```

Reply:
you mean I forgot the wait on version?